Your pull request does not meet our code style rules. Pylint summary:
```
************* Module dpd
testsuite/python/dpd.py:431:8: W0612: Unused variable 'F_max' (unused-variable)
```

You can generate these warnings with `maintainer/CI/fix_style.sh`. This is the same command that I have executed to generate the log above.

--- a/src/core/Observable_stat.cpp
+++ b/src/core/Observable_stat.cpp
@@ -46,6 +46,11 @@ Observable_stat::Observable_stat(std::size_t chunk_size, std::size_t n_bonded,
 #else
   constexpr std::size_t n_vs = 0;
 #endif
+#ifdef DPD
+  constexpr std::size_t n_dpd = 1;
+#else
+  constexpr std::size_t n_dpd = 0;
+#endif
   auto const n_non_bonded = get_non_bonded_offset(max_type, max_type) + 1ul;
   constexpr std::size_t n_ext_fields = 1;  // reduction over all fields
   constexpr std::size_t n_kinetic_lin = 1; // linear kinetic contribution
@@ -53,7 +58,7 @@ Observable_stat::Observable_stat(std::size_t chunk_size, std::size_t n_bonded,
 
   auto const n_elements = n_kinetic_lin + n_kinetic_rot + n_bonded +
                           2ul * n_non_bonded + n_coulomb + n_dipolar + n_vs +
-                          n_ext_fields;
+                          n_ext_fields + n_dpd;
   m_data = std::vector<double>(m_chunk_size * n_elements);
 
   // spans for the different contributions
@@ -63,8 +68,8 @@ Observable_stat::Observable_stat(std::size_t chunk_size, std::size_t n_bonded,
   coulomb = std::span<double>(bonded.end(), n_coulomb * m_chunk_size);
   dipolar = std::span<double>(coulomb.end(), n_dipolar * m_chunk_size);
   virtual_sites = std::span<double>(dipolar.end(), n_vs * m_chunk_size);
-  external_fields =
-      std::span<double>(virtual_sites.end(), n_ext_fields * m_chunk_size);
+  dpd = std::span<double>(virtual_sites.end(), n_dpd * m_chunk_size);
+  external_fields = std::span<double>(dpd.end(), n_ext_fields * m_chunk_size);
   non_bonded_intra =
       std::span<double>(external_fields.end(), n_non_bonded * m_chunk_size);
   non_bonded_inter =

--- a/src/core/Observable_stat.hpp
+++ b/src/core/Observable_stat.hpp
@@ -82,6 +82,8 @@ public:
   std::span<double> dipolar;
   /** Contribution from virtual sites (accumulated). */
   std::span<double> virtual_sites;
+  /** Contribution from DPD. */
+  std::span<double> dpd;
   /** Contribution from external fields (accumulated). */
   std::span<double> external_fields;
   /** Contribution(s) from non-bonded intramolecular interactions. */

--- a/src/core/dpd.cpp
+++ b/src/core/dpd.cpp
@@ -156,6 +156,11 @@ static auto dpd_viscous_stress_local() {
   return stress;
 }
 
+Utils::Vector9d dpd_pressure_local() {
+  auto const local_stress = dpd_viscous_stress_local();
+  return -Utils::flatten(local_stress);
+}
+
 /**
  * @brief Viscous stress tensor of the DPD interaction.
  *
@@ -174,7 +179,7 @@ static auto dpd_viscous_stress_local() {
 Utils::Vector9d dpd_stress(boost::mpi::communicator const &comm) {
   auto const &box_geo = *System::get_system().box_geo;
   auto const local_stress = dpd_viscous_stress_local();
-  std::remove_const_t<decltype(local_stress)> global_stress;
+  std::remove_const_t<decltype(local_stress)> global_stress{};
 
   boost::mpi::reduce(comm, local_stress, global_stress, std::plus<>(), 0);
 

--- a/src/core/dpd.hpp
+++ b/src/core/dpd.hpp
@@ -51,5 +51,10 @@ dpd_pair_force(Particle const &p1, Particle const &p2, DPDThermostat const &dpd,
                BoxGeometry const &box_geo, IA_parameters const &ia_params,
                Utils::Vector3d const &d, double dist, double dist2);
 Utils::Vector9d dpd_stress(boost::mpi::communicator const &comm);
+/**
+ * @brief Local contribution to the pressure tensor.
+ * Needs to be rescaled by the box volume.
+ */
+Utils::Vector9d dpd_pressure_local();
 
 #endif // DPD

--- a/src/core/pressure.cpp
+++ b/src/core/pressure.cpp
@@ -26,6 +26,7 @@
 #include "Particle.hpp"
 #include "ParticleRange.hpp"
 #include "bonded_interactions/bonded_interaction_data.hpp"
+#include "dpd.hpp"
 #include "electrostatics/coulomb.hpp"
 #include "magnetostatics/dipoles.hpp"
 #include "nonbonded_interactions/nonbonded_interaction_data.hpp"
@@ -114,6 +115,10 @@ std::shared_ptr<Observable_stat> System::calculate_pressure() {
     std::ranges::copy(Utils::flatten(vs_pressure),
                       obs_pressure.virtual_sites.begin());
   }
+#endif
+
+#ifdef DPD
+  std::ranges::copy(dpd_pressure_local(), obs_pressure.dpd.begin());
 #endif
 
   obs_pressure.rescale(volume);

--- a/src/python/espressomd/analyze.py
+++ b/src/python/espressomd/analyze.py
@@ -652,6 +652,8 @@ class Analysis(ScriptInterfaceHelper):
             reduction(out, "dipolar")
         if has_features("VIRTUAL_SITES"):
             reduction(out, "virtual_sites")
+        if has_features("DPD"):
+            reduction(out, "dpd")
 
         if dim == 1 or calc_sp:
             return out

--- a/src/script_interface/analysis/ObservableStat.cpp
+++ b/src/script_interface/analysis/ObservableStat.cpp
@@ -142,6 +142,15 @@ static auto get_summary(::System::System const &system,
   }
 #endif // VIRTUAL_SITES
 
+#ifdef DPD
+  {
+    auto const values = get_obs_contribs(obs.dpd);
+    for (std::size_t i = 0ul; i < values.size(); ++i) {
+      dict["dpd," + std::to_string(i)] = values[i];
+    }
+  }
+#endif // DPD
+
   return dict;
 }
 

--- a/testsuite/python/analyze_energy.py
+++ b/testsuite/python/analyze_energy.py
@@ -174,6 +174,8 @@ class AnalyzeEnergy(ut.TestCase):
         self.assertAlmostEqual(energy["non_bonded"], 1., delta=1e-7)
         self.assertAlmostEqual(energy["non_bonded"], 0.5 * sum(
             [self.system.analysis.particle_energy(p) for p in self.system.part.all()]), delta=1e-7)
+        self.assertAlmostEqual(energy["virtual_sites"], 0., delta=1e-7)
+        self.assertAlmostEqual(energy["dpd"], 0., delta=1e-7)
         # two bonds
         p1.add_bond((self.harmonic, p0))
         energy = self.system.analysis.energy()

--- a/testsuite/python/dpd.py
+++ b/testsuite/python/dpd.py
@@ -44,6 +44,7 @@ class DPDThermostat(ut.TestCase):
         self.system.constraints.clear()
         self.system.thermostat.turn_off()
         self.system.integrator.set_vv()
+        self.system.non_bonded_inter.reset()
 
     def test_01__rng(self):
         """Test for RNG consistency."""
@@ -394,9 +395,11 @@ class DPDThermostat(ut.TestCase):
 
             dpd_obs = espressomd.observables.DPDStress()
             obs_stress = dpd_obs.calculate()
+            pressure = system.analysis.pressure_tensor()["dpd"]
 
             np.testing.assert_array_almost_equal(np.copy(dpd_stress), stress)
             np.testing.assert_array_almost_equal(np.copy(obs_stress), stress)
+            np.testing.assert_array_almost_equal(np.copy(pressure), -stress)
 
     def test_momentum_conservation(self):
         r_cut = 1.0

